cdci_data_analysis/analysis/instrument.py

```diff
@@ -120,11 +125,22 @@ def set_pars_from_dic(self, par_dic, verbose=False):
             query_name = self.get_product_query_name(product_type)
             query_obj = self.get_query_by_name(query_name)
             for par in query_obj._parameters_list:
-                par.set_from_form(par_dic, verbose=verbose)
+                # this is required because in some cases a parameter is set without a name (eg UserCatalog)
```

Comment:
is it not possible to combine 3 loops in one?

Reply:
yes, totally makes sense